src/Sylius/Component/Core/OrderProcessing/OrderPaymentProcessor.php

```diff
@@ -62,7 +62,7 @@ public function process(BaseOrderInterface $order)
             return;
         }
 
-        $newPayment = $order->getLastNewPayment();
+        $newPayment = $order->getLastCartPayment();
```

Comment:
`fail` and `cancel` transition on payment is triggered after the order was placed...